File: ios/App/App/WidgetSnapshotStore.swift
import Foundation

enum WidgetConstants {
    static let appGroupId = "group.co.slidepress.app"
    static let snapshotKey = "widgetSnapshot"
}

enum WidgetSnapshotStore {
    static var defaults: UserDefaults? {
        UserDefaults(suiteName: WidgetConstants.appGroupId)
    }

    static func saveSnapshotJson(_ json: String) {
        defaults?.set(json, forKey: WidgetConstants.snapshotKey)
    }

    static func loadSnapshotJson() -> String? {
        defaults?.string(forKey: WidgetConstants.snapshotKey)
    }

    static func clearSnapshot() {
        defaults?.removeObject(forKey: WidgetConstants.snapshotKey)
    }
}

File: android/app/src/main/java/co/slidepress/app/widget/WidgetSnapshotStore.kt
package co.slidepress.app.widget

import android.content.Context

object WidgetSnapshotStore {
    fun saveSnapshotJson(context: Context, json: String) {
        context
            .getSharedPreferences(WidgetConstants.PREFS_NAME, Context.MODE_PRIVATE)
            .edit()
            .putString(WidgetConstants.SNAPSHOT_KEY, json)
            .commit()
    }

    fun loadSnapshotJson(context: Context): String? {
        return context
            .getSharedPreferences(WidgetConstants.PREFS_NAME, Context.MODE_PRIVATE)
            .getString(WidgetConstants.SNAPSHOT_KEY, null)
    }

    fun clearSnapshot(context: Context) {
        context
            .getSharedPreferences(WidgetConstants.PREFS_NAME, Context.MODE_PRIVATE)
            .edit()
            .remove(WidgetConstants.SNAPSHOT_KEY)
            .commit()
    }
}
